### ISSUE TITLE
for(;;) whitespace issue

### ISSUE THREAD

**Issue by benny-medflyt** [palantir/tslint#2934]
### Bug Report

- __TSLint version__: 5.4.3
- __TypeScript version__: 2.3.4
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
for(; ;) {
    // Do Nothing
}
```

with `tslint.json` configuration:

```json
{
    "extends": ["tslint:latest"],
    "rules": {
        "whitespace": [true,
            "check-separator"
        ],
    }
}
```

#### Actual behavior

```
ERROR: 1:13  whitespace  missing whitespace
```

#### Expected behavior

Should successfully pass tlint.

tslint wants the code to look like `for (; ; )` (note the extra whitespace). This is ugly, and is also goes against the default way that Visual Studio Code formats it

**Comment by timocov** [comment]
Duplicate of #2382
